node/pkg/altpub/alternate_pub.go

```diff
@@ -0,0 +1,534 @@
+		return
+	}
+	r.Header.Add("Content-Type", "application/octet-stream")
+
+	start := time.Now()
+	resp, err := client.Do(r)
+	if err != nil {
+		ap.logger.Error("Failed to post request", zap.Error(err))
+		if strings.Contains(err.Error(), "connection refused") {
+			requestFailed.WithLabelValues(req.ep.label, "connection_refused").Inc()
+		} else {
+			requestFailed.WithLabelValues(req.ep.label, "post_failed").Inc()
+		}
+		return
+	}
+	stop := time.Now()
+	resp.Body.Close()
+
+	// Peg the appropriate metric, based on the request result.
+	if resp.StatusCode < 300 {
+		requestSuccess.WithLabelValues(req.ep.label).Inc()
+		postTime.WithLabelValues(req.ep.label).Observe(float64(stop.Sub(start).Milliseconds()))
+	} else {
+		reason := http.StatusText(resp.StatusCode)
+		if reason == "" {
+			reason = fmt.Sprintf("status_code_%d", resp.StatusCode)
+		}
+		requestFailed.WithLabelValues(req.ep.label, reason).Inc()
+		ap.logger.Error("Failed to post request", zap.String("endpoint", req.ep.label), zap.Int("statusCode", resp.StatusCode), zap.String("statusText", reason))
+	}
+}
+
+// createUrls is a helper for the endpoint that formats the URLs for each request type.
+// NOTE: Initially there is only one request type, but that may change later.
+func (ep *Endpoint) createUrls() {
+	ep.signedObservationUrl = ep.baseUrl + "/SignedObservationBatch"
+}
+
+// shouldPublish is a helper for the endpoint that checks to see if the chain passed in is in the set of chains enabled for the endpoint.
+func (ep *Endpoint) shouldPublish(chainId vaa.ChainID) bool {
+	return ep.enabledChains.shouldPublish(chainId)
+}
+
+// batchWorker is the entrypoint for a per-endpoint worker that batches observations based on the batching interval.
+// This is only created if batching is enabled for the endpoint.
+func (w *BatchWorker) batchWorker(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+			if err := w.handleBatch(ctx); err != nil {
+				if errors.Is(err, context.Canceled) {
+					return nil
+				}
+
+				return fmt.Errorf("handleBatch failed: %w", err)
+			}
+		}
+	}
+}
+
+// handleBatch waits up to the delay interval to batch requests. It then formats a request and posts it to the httpWorkerChan.
+func (w *BatchWorker) handleBatch(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Duration(w.ep.delay))
+	defer cancel()
+
+	observations, err := common.ReadFromChannelWithTimeout(ctx, w.ep.obsvBatchChan, p2p.MaxObservationBatchSize)
+	if err != nil && !errors.Is(err, context.DeadlineExceeded) {
+		return fmt.Errorf("failed to read observations from the internal observation batch channel: %w", err)
+	}
+
+	if len(observations) != 0 {
+		batch := gossipv1.SignedObservationBatch{
+			Addr:         w.ap.guardianAddr,
+			Observations: observations,
+		}
+
+		data, err := proto.Marshal((&batch))
+		if err != nil {
+			panic("failed to marshal batch")
```

Comment:
Same as above, I think normal error handling is better here.

Reply:
Same comment as above.